src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -17,6 +17,44 @@ use crate::{error::MathError, integer::PolyOverZ, integer_mod_q::modulus::Modulu
 use flint_sys::fmpz_mod_poly::{fmpz_mod_poly_init, fmpz_mod_poly_set_fmpz_poly};
 use std::{mem::MaybeUninit, str::FromStr};
 
+impl From<(&PolyOverZ, &Modulus)> for PolyOverZq {
+    /// Create a [`PolyOverZq`] from a [`PolyOverZ`] and [`Modulus`].
+    ///
+    /// Parameters:
+    /// - `poly_modulus_tuple`: A tuple of the polynomial and the modulus.
+    ///
+    /// Examples:
+    /// ``` rust
+    /// use qfall_math::integer_mod_q::{PolyOverZq, Modulus};
+    /// use qfall_math::integer::PolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZ::from_str("4  0 1 102 3").unwrap();
+    /// let modulus = Modulus::from_str("100").unwrap();
```

Comment:
We should probably add a TryFrom<impl Into<Z>> to our implementation, as using a from_str in doc comments is not really that nice